test/checkout.int.t.sol

```diff
@@ -1119,8 +1171,150 @@ contract Checkout_swapFromSupplierWithoutFeeSpecialOrder is Checkout {
+     * Fulfillment should result in all of supplier2's 2019 removal being used, and .5 tonne of supplier2's 2020 removal
+     */
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier,
+        amount: 1 ether,
+        vintage: 2018
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier2,
+        amount: 1 ether,
+        vintage: 2018
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier2,
+        amount: 1 ether,
+        vintage: 2019
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier2,
+        amount: 1 ether,
+        vintage: 2020
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier3,
+        amount: 1 ether,
+        vintage: 2019
+      })
+    );
+    uint256 purchaseAmount = _market.calculateCheckoutTotalWithoutFee({
+      amount: certificateAmount,
+      priceMultiple: _priceMultiple
+    });
+    _market.grantRole({role: _market.MARKET_ADMIN_ROLE(), account: owner});
+    vm.prank(_namedAccounts.admin);
+    _bpNori.deposit(owner, abi.encode(purchaseAmount));
+    vm.prank(owner);
+    _bpNori.approve(address(_market), purchaseAmount);
+  }
+
+  function test_basicFulfillment() external {
+    vm.prank(owner);
+    vm.recordLogs();
+    _market.swapWithoutFeeSpecialOrder(
+      owner,
+      owner,
+      certificateAmount,
+      customFee,
+      _priceMultiple,
+      _namedAccounts.supplier2,
+      vintages
+    );
+
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    // In this test scenario, the `CreateCertificate` event index is 18 because multiple suppliers are getting
+    // paid so we use the event selector to find the event index.
+    uint256 createCertificateEventIndex;
+    for (uint256 i = 0; i < entries.length; ++i) {
+      if (entries[i].topics[0] == CREATE_CERTIFICATE_EVENT_SELECTOR) {
+        createCertificateEventIndex = i;
+        break;
+      }
+    }
+    assertEq(
+      entries[createCertificateEventIndex].topics[0],
+      CREATE_CERTIFICATE_EVENT_SELECTOR
+    );
+    assertEq(
+      entries[createCertificateEventIndex].topics[1],
+      bytes32(uint256(uint160(address(owner))))
+    );
+    assertEq(
+      entries[createCertificateEventIndex].topics[2],
+      bytes32(uint256(uint256(0)))
+    );
+    assertEq(
+      entries[createCertificateEventIndex].topics[3],
+      bytes32(uint256(uint160(address(_bpNori))))
+    );
+    (
+      address from,
+      uint256 eventCertificateAmount,
+      uint256[] memory removalIds,
+      uint256[] memory removalAmounts,
+      uint256 priceMultiple,
+      uint256 noriFeePercentage
+    ) = abi.decode(
+        entries[createCertificateEventIndex].data,
+        (address, uint256, uint256[], uint256[], uint256, uint256)
+      );
```

Comment:
Should we just use expect emit instead of recorded logs for this test?

Reply:
Get recorded logs would be more useful if this test tested more than just the one event